### PR TITLE
Add ssh public key for Inband management.

### PR DESCRIPTION
**Pull request by lanchongyizu**
Public key is required for ssh connection for Inband management, and private key is optional.
This is related with ODR-972.

@RackHD/corecommitters @panpan0000 @iceiilin @benbp

### CODE REVIEW

**Comment by lanchongyizu** [comment]
Jenkins: depends on https://github.com/RackHD/on-tasks/pull/369

**Comment by iceiilin** [comment]
:+1:

File: lib/serializables/v1/ssh.js [hunk 1]
```diff
@@ -62,6 +62,9 @@ function SshFactory (Promise, Serializable, encryption) {
         if (this.password) {
             this.password = 'REDACTED';
         }
+        if (this.publicKey) {
+            this.publicKey = 'REDACTED';
+        }
```
Comment (johren) [inline lib/serializables/v1/ssh.js]: Public keys do not need to be encrypted or redacted.  These can be safely revealed to anyone.

File: lib/serializables/v1/ssh.js [hunk 2]
```diff
@@ -76,6 +79,9 @@ function SshFactory (Promise, Serializable, encryption) {
         if (self.password) {
             self.password = encryption.encrypt(self.password);
         }
+        if (self.publicKey) {
+            self.publicKey = encryption.encrypt(self.publicKey);
+        }
```
Comment (johren) [inline lib/serializables/v1/ssh.js]: Public key does not need to be encrypted.

File: spec/lib/serializables/v1/ssh-spec.js [hunk 3]
```diff
@@ -32,13 +32,23 @@ describe('Ssh Serializable V1', function () {
...
+        it('should redact encrypted publicKey fields', function() {
+            return this.subject.serialize(
+                {
+                    host: 'fake-host',
+                    user: 'fake-user',
+                    publicKey: encryption.encrypt('fake-public-key')
+                }
+            ).should.eventually.have.property('publicKey').that.equals('REDACTED');
+        });
+
```
Comment (johren) [inline spec/lib/serializables/v1/ssh-spec.js]: Public key does not need to be encrypted or redacted.

File: spec/lib/serializables/v1/ssh-spec.js [hunk 4]
```diff
@@ -114,6 +124,19 @@ describe('Ssh Serializable V1', function () {
...
+                    host: 'fake-host',
+                    user: 'fake-user',
+                    publicKey: 'fake-public-key',
+                    privateKey: 'fake-private-key'
+                }
+            ).should.eventually.have.property(
+                'publicKey'
+            ).and.not.equal('fake-public-key');
+        });
+
```
Comment (johren) [inline spec/lib/serializables/v1/ssh-spec.js]: Public key does not need to be encrypted.

**Pull request merged**